### ISSUE TITLE
TopicMerge sets unexpected @xml:lang attribute to topic.

### ISSUE THREAD

**Issue by ToshihikoMakita** [dita-ot/dita-ot#1476]
I'm facing a serious problem found in DITA-OT 1.7.3. The problem is that DITA-OT TopicMerge sets unexpected `@xml:lang` attribute to topic. Here is a example.

[bookmap: sample_ja.bookmap]
```xml
<bookmap xml:lang="ja" id="GUID-BC13DC41-C380-4E24-9047-B0E6018F4D86">
  <booktitle id="GUID-877192EE-6DCD-4695-AD85-E1F54ED545E5"> 
    <mainbooktitle>テスト</mainbooktitle> 
  </booktitle> 
  ...
  <chapter format="dita" href="sample.xml" scope="local" navtitle="サンプル"/>
</bookmap>
```

[sample.xml]
```xml
<concept id="concept_D941EF43C15A40D0BF155B7C5B8003F0">
  <title>コンセプトサンプル</title>
</concept>
```

[sample_ja_MERGED.xml]
```xml
<concept ditaarch:DITAArchVersion="1.2" domains="(topic concept)                            (topic hi-d)                             (topic ut-d)                             (topic indexing-d)                            (topic hazard-d)                            (topic abbrev-d)                            (topic pr-d)                             (topic sw-d)                            (topic ui-d)    " 
         class="- topic/topic concept/concept " xtrf="D:\DITA-OT1.7.3\samples\sample.xml" xtrc="concept:1;3:56"
         xml:lang="en" oid="concept_D941EF43C15A40D0BF155B7C5B8003F0" id="unique_1">
  <title class="- topic/title " xtrf="D:\DITA-OT1.7.3\samples\sample.xml" xtrc="title:1;4:8">コンセプトサンプル</title>
...
</concept>
```

As shown above TopicMerge sets topic/@xml:lang="en" as default value.

Until now we use DITA-OT 1.5.3 and our stylesheet honors `bookmap/@xml:lang`. The stylesheet inherit `bookmap/@xml:lang` to the topic and element level. So according to this situation, `bookmap/@xml:lang="ja"` should be honored.

Are there any TopicMerge option to avoid setting `@xml:lang="en"` as default value?

**Comment by ToshihikoMakita** [comment]
Any progress on this issue?

This will not affect following user.

- CMS user (because CMS forces to specify topic/@xml:lnag explicitly.)
- English user
- PDF2 plug-in user (It only use map/@xml:lang only.)

But there are lot of DITA instances that does not explicitly specify topic/@xml:lnag. In this case user believes that map/@xml:lnag is used for topic/@xml:lang.

I'm developing multiple language stylesheet per user request. In this stylesheet the style is controlled by @xml:lang attribute of the element. So applying topic/@xml:lang="en" is very headache problem.

I'm waiting for your reply.

Regards,

**Comment by robander** [comment]
Just noticed the long-ago question on this. I think it would be appropriate, especially for PDF, if our processing default was "whatever language is specified for the map", and if nothing is specified on the map it can fall back to 'en' as today. I think this would mean updating the merged file so that it only adds a default to the map, and updating processing for any topic context so that it falls back to the value on the map.

**Comment by ToshihikoMakita** [comment]
Thank you for your comment.
Anyway I am very appreciated if "TopicMerge" changes current behavior. This problem is not meaningful for commercial CMS users because CMS strictly controls @xml:lang attribute for every topic. But there are lots of user topics that have no explicit @xml:lang. In this case they want to treat map/@xml-lang as representative language.

**Comment by jelovirt** [comment]
Fixed in #2356

**Comment by ToshihikoMakita** [comment]
Thank you very much!
